### PR TITLE
After click on a category a list of needed recipes is shown

### PR DESCRIPTION
**Pull request by MartaLiutyk**
On click on Categorie we have  a list of pecipes of this categorie

### CODE REVIEW

File: app/src/main/java/bubbleheads/buy_cook/CategoriesAdapter.java [hunk 1]
```diff
@@ -1,62 +1,70 @@
...
+import bubbleheads.buy_cook.CategoriesAdapter.MyViewHolder;
+
+import org.w3c.dom.Text;
 
 import java.util.ArrayList;
 import java.util.List;
 
-/**
- * Created by melod on 29.03.2017.
- */
```
Comment (Pastornak) [inline app/src/main/java/bubbleheads/buy_cook/CategoriesAdapter.java]: That shouldn`t be here

File: app/src/main/java/bubbleheads/buy_cook/CategoriesAdapter.java [hunk 2]
```diff
@@ -1,62 +1,70 @@
...
 import android.view.ViewGroup;
 import android.widget.BaseAdapter;
 import android.widget.TextView;
+import android.support.v7.widget.RecyclerView.Adapter;
+import android.support.v7.widget.RecyclerView.ViewHolder;
+import android.widget.Toast;
+
+import bubbleheads.buy_cook.CategoriesAdapter.MyViewHolder;
+
+import org.w3c.dom.Text;
```
Comment (Pastornak) [inline app/src/main/java/bubbleheads/buy_cook/CategoriesAdapter.java]: What an import is this?

File: app/src/main/res/layout/categories_cell.xml [hunk 3]
```diff
@@ -1,15 +1,13 @@
 <?xml version="1.0" encoding="utf-8"?>
 <LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
-    android:orientation="vertical" android:layout_width="match_parent"
+    android:orientation="vertical"
+    android:layout_width="match_parent"
     android:layout_height="wrap_content">
```
Comment (Pastornak) [inline app/src/main/res/layout/categories_cell.xml]: No empty line there (or add at the end of TextView)

File: app/src/main/java/bubbleheads/buy_cook/CategoriesAdapter.java [hunk 4]
```diff
@@ -1,62 +1,70 @@
...
-        public TextView text;
+public class CategoriesAdapter extends Adapter<MyViewHolder> {
+    private final CategoriesFragment categoriesFragment;
 
-        public View itemView;
+    protected static class MyViewHolder extends ViewHolder {
 
+        private final TextView text;
+        private final View view;
+        private final RecyclerView rv;
```
Comment (Pastornak) [inline app/src/main/java/bubbleheads/buy_cook/CategoriesAdapter.java]: Do you really need this field?

File: app/src/main/java/bubbleheads/buy_cook/CategoriesFragment.java [hunk 5]
```diff
@@ -8,45 +8,30 @@
...
 
 import java.util.ArrayList;
 
-/**
- * Created by melod on 29.03.2017.
- */
-
 public class CategoriesFragment extends Fragment {
-    private ArrayList<Category> categoryArrayList = new ArrayList<>();
     private RecyclerView recyclerView;
```
Comment (Pastornak) [inline app/src/main/java/bubbleheads/buy_cook/CategoriesFragment.java]: Can be transformed into local variable

File: app/src/main/java/bubbleheads/buy_cook/CategoriesFragment.java [hunk 6]
```diff
@@ -8,45 +8,30 @@
...
     @Override
     public void onResume() {
         super.onResume();
-        prepareCategoriesData();
-        CategoriesAdapter mAdapter = new CategoriesAdapter(this);
-        recyclerView.setLayoutManager(new LinearLayoutManager(getContext(), LinearLayoutManager.VERTICAL, false));
-        recyclerView.setAdapter(mAdapter);
-        mAdapter.notifyDataSetChanged();
-
     }
```
Comment (Pastornak) [inline app/src/main/java/bubbleheads/buy_cook/CategoriesFragment.java]: Empty line

File: app/src/main/java/bubbleheads/buy_cook/RecipesBookAdapter.java [hunk 7]
```diff
@@ -13,30 +14,25 @@
...
- */
-
-public class RecipesBookAdapter extends BaseAdapter implements Filterable {
-    private Context context;
+public final class RecipesBookAdapter extends BaseAdapter implements Filterable {
+    private final Context context;
     private ArrayList<Recipe> recipes;
-    private ArrayList<Recipe> backupRecipes;
+    private final ArrayList<Recipe> backupRecipes;
     private RecipesBookAdapter.ValueFilter valueFilter;
```
Comment (Pastornak) [inline app/src/main/java/bubbleheads/buy_cook/RecipesBookAdapter.java]: Final fields go first

File: app/src/main/java/bubbleheads/buy_cook/RecipesBookFragment.java [hunk 8]
```diff
@@ -12,55 +12,65 @@
...
 
-/**
- * Created by melod on 26.03.2017.
- */
 public class RecipesBookFragment extends Fragment {
     private GridView collectionView;
     private ArrayList<Recipe> recipes;
     private SearchView findRecipe;
     private RecipesBookAdapter recipesBookAdapter;
+    private Category chosenCategory = null;
```
Comment (Pastornak) [inline app/src/main/java/bubbleheads/buy_cook/RecipesBookFragment.java]: Some fields can be transfered to local final variables

File: app/src/main/res/layout/categories_fragment.xml [hunk 9]
```diff
@@ -1,15 +1,12 @@
 <?xml version="1.0" encoding="utf-8"?>
-<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
     android:orientation="vertical">
```
Comment (Pastornak) [inline app/src/main/res/layout/categories_fragment.xml]: Empty line (or add at the end of RecycleView)

File: app/src/main/java/bubbleheads/buy_cook/CategoriesAdapter.java (hunk 1 above)
Comment (MartaLiutyk) [inline app/src/main/java/bubbleheads/buy_cook/CategoriesAdapter.java]: it is already deleted (red , OK ?)

File: app/src/main/java/bubbleheads/buy_cook/CategoriesAdapter.java (hunk 4 above)
Comment (MartaLiutyk) [reply app/src/main/java/bubbleheads/buy_cook/CategoriesAdapter.java]: yes.

File: app/src/main/java/bubbleheads/buy_cook/CategoriesFragment.java (hunk 6 above)
Comment (MartaLiutyk) [inline app/src/main/java/bubbleheads/buy_cook/CategoriesFragment.java]: it`s red. DELETED

Comment (Pastornak) [inline app/src/main/java/bubbleheads/buy_cook/CategoriesFragment.java]: It`s not red. It`s white.

File: app/src/main/java/bubbleheads/buy_cook/CategoriesAdapter.java [hunk 10]
```diff
@@ -49,6 +35,15 @@ public MyViewHolder onCreateViewHolder(final ViewGroup parent, final int viewTyp
     public void onBindViewHolder(final MyViewHolder holder, final int position) {
         final TextView categoryName = (TextView) holder.itemView.findViewById(R.id.categories_name);
         categoryName.setText(Category.values()[position].getName());
+        categoryName.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Toast.makeText(view.getContext(), Category.values()[position].getName(), Toast.LENGTH_SHORT).show();
```
Comment (ovovchak) [inline app/src/main/java/bubbleheads/buy_cook/CategoriesAdapter.java]: isn't this line too long?

Comment (ovovchak) [inline app/src/main/java/bubbleheads/buy_cook/CategoriesAdapter.java]: inside this method you twice use Category.values()[position] - this is called code duplication. Extract this as local varibal final Category category =  Category.values()[position]

File: app/src/main/java/bubbleheads/buy_cook/CategoriesFragment.java [hunk 11]
```diff
@@ -8,28 +8,28 @@
...
 public class CategoriesFragment extends Fragment {
-    private RecyclerView recyclerView;
 
     @Override
     public View onCreateView(final LayoutInflater inflater,
                              final ViewGroup container,
                              final Bundle savedInstanceState) {
         final View view = inflater.inflate(R.layout.categories_fragment, container, false);
-        recyclerView = (RecyclerView) view.findViewById(R.id.recycler_view);
+        RecyclerView recyclerView = (RecyclerView) view.findViewById(R.id.recycler_view);
```
Comment (ovovchak) [inline app/src/main/java/bubbleheads/buy_cook/CategoriesFragment.java]: final?

File: app/src/main/java/bubbleheads/buy_cook/RecipesBookAdapter.java [hunk 12]
```diff
@@ -16,8 +16,8 @@
 
 public final class RecipesBookAdapter extends BaseAdapter implements Filterable {
     private final Context context;
-    private ArrayList<Recipe> recipes;
     private final ArrayList<Recipe> backupRecipes;
```
Comment (ovovchak) [inline app/src/main/java/bubbleheads/buy_cook/RecipesBookAdapter.java:19]: final fields should be close to final and not final to not final

File: app/src/main/java/bubbleheads/buy_cook/RecipesBookFragment.java [hunk 13]
```diff
@@ -42,13 +55,12 @@ public void onItemClick(final AdapterView<?> adapterView, final View view,
...
                 recipesBookAdapter.getRecipes().get(position).changeFavorite();
                 recipesBookAdapter.notifyDataSetChanged();
-
             }
         });
     }
 
     private void setUpSearching(final View view) {
-        findRecipe = (SearchView) view.findViewById(R.id.searchRecipe);
+        SearchView findRecipe = (SearchView) view.findViewById(R.id.searchRecipe);
```
Comment (ovovchak) [inline app/src/main/java/bubbleheads/buy_cook/RecipesBookFragment.java]: final?

File: app/src/main/java/bubbleheads/buy_cook/RecipesBookFragment.java [hunk 14]
```diff
@@ -13,24 +13,37 @@
...
+        return result;
+    }
+
+    public void setChosenCategory(final Category category) {
+        this.chosenCategory = chosenCategory;
+    }
+
     private void setUpCollection(final View view) {
-        collectionView = (GridView) view.findViewById(R.id.collection_view);
+        GridView collectionView = (GridView) view.findViewById(R.id.collection_view);
```
Comment (ovovchak) [inline app/src/main/java/bubbleheads/buy_cook/RecipesBookFragment.java]: final ?

**Review by ovovchak — CHANGES_REQUESTED** [review]

**Pull request merged**